libbeat/_meta/config/logging.reference.yml.tmpl

```diff
@@ -67,3 +67,51 @@ logging.files:
   # Rotate existing logs on startup rather than appending them to the existing
   # file. Defaults to true.
   # rotateonstartup: true
+
+#=============================== Events Logging ===============================
+# Some outputs will log raw events on errors like indexing errors in the
+# Elasticsearch output, to prevent logging raw events (that may contain
+# sensitive information) together with other log messages, a different
+# log file, only for log entries containing raw events, is used. It will
+# use the same level, selectors and all other configurations from the
+# default logger, but it will have it's own file configuration.
+#
+# Having a different log file for raw events also prevents event data
+# from drowning out the regular log files.
+#
+# IMPORTANT: No matter the default logger output configuration, raw events
+# will **always** be logged to a file configured by `logging.event_data.files`.
+
+# logging.event_data:
+# Logging to rotating files. Set logging.to_files to false to disable logging to
+# files.
+#logging.event_data.to_files: true
```

Comment:
```suggestion
# logging.event_data:
# Logging to rotating files. Set logging.event_data.to_files to false to disable logging event data to
# files.
#logging.event_data.to_files: true
```

Also to double check, if you turn this off the event data goes into the main log files like it did before this change?

Reply:
> Also to double check, if you turn this off the event data goes into the main log files like it did before this change?

Tried it, it does not.

Reply:
Yep, it does not. It's lost forever.

The logger core that does the filtering/routing of the log entries is always instantiated. Turning off the logger outputs instantiates a functional logger (from the code perspective) that just discards all log entries.

Reply:
That's not what I observed, I still saw the event logs going into the files, like this setting had no observable effect at all.

Reply:
That was an issue with `elastic-agent-libs/logp`, I fixed it on https://github.com/elastic/elastic-agent-libs/pull/204 and already added the updated lib to this PR.